### PR TITLE
Add semantics for Sliders

### PR DESCRIPTION
**Pull request by abarth**
Also, make SemanticsOwner into a real class and use it instead of a static in
several places.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/material/slider.dart [hunk 1]
```diff
@@ -417,4 +421,32 @@ class _RenderSlider extends RenderConstrainedBox {
...
     canvas.drawCircle(thumbCenter, thumbRadius + thumbRadiusDelta, thumbPaint);
   }
+
+  @override
+  bool get hasSemantics => isInteractive;
+
+  @override
+  Iterable<SemanticAnnotator> getSemanticAnnotators() sync* {
+    yield (SemanticsNode semantics) {
+      if (isInteractive)
```
Comment (Hixie) [inline packages/flutter/lib/src/material/slider.dart]: eventually we should presumably expose the value of the slider?

File: packages/flutter/lib/src/rendering/binding.dart [hunk 2]
```diff
@@ -133,13 +133,19 @@ abstract class RendererBinding extends BindingBase implements SchedulerBinding,
...
+      ensureSemantics();
       mojom.SemanticsServerStub server = new mojom.SemanticsServerStub.fromEndpoint(endpoint);
-      server.impl = new SemanticsServer();
+      server.impl = new SemanticsServer(semanticsOwner: pipelineOwner.semanticsOwner);
     });
   }
 
+  void ensureSemantics() {
+    if (pipelineOwner.semanticsOwner == null)
+      renderView.scheduleInitialSemantics();
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/binding.dart]: o_O

Not sure what to suggest instead. It's a bit weird to rely so explicitly on one object changing another object and having a third object be responsible for making sure the first gets called when the second needs it...

File: packages/flutter/lib/src/rendering/object.dart [hunk 3]
```diff
@@ -624,7 +624,7 @@ class _RootSemanticsFragment extends _InterestingSemanticsFragment {
     assert(parentSemantics == null);
     owner._semantics ??= new SemanticsNode.root(
       handler: owner is SemanticActionHandler ? owner as dynamic : null,
-      owner: owner.owner
+      owner: owner.owner.semanticsOwner
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: ...

Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: not sure whether to suggest that we rename the two `owner`s, or whether to suggest that `semanticsOwner` should be called `owner`.

File: packages/flutter/lib/src/widgets/semantics_debugger.dart [hunk 4]
```diff
@@ -366,10 +377,13 @@ class _SemanticsDebuggerListener extends ChangeNotifier implements mojom.Semanti
...
       return;
     if (vx.abs() > vy.abs()) {
-      if (vx.sign < 0)
+      if (vx.sign < 0) {
+        _performAction(position, SemanticAction.decrease);
         _performAction(position, SemanticAction.scrollLeft);
-      else
+      } else {
+        _performAction(position, SemanticAction.increase);
         _performAction(position, SemanticAction.scrollRight);
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart]: doing both seems a bit weird but ok

File: packages/flutter/test/rendering/reattach_test.dart [hunk 5]
```diff
@@ -136,7 +136,8 @@ void main() {
   test('objects can be detached and re-attached: semantics', () {
     TestTree testTree = new TestTree();
     TestSemanticsListener listener = new TestSemanticsListener();
-    SemanticsNode.addListener(listener);
+    renderer.ensureSemantics();
+    renderer.pipelineOwner.semanticsOwner.addListener(listener);
```
Comment (Hixie) [inline packages/flutter/test/rendering/reattach_test.dart]: that's nicer than owner.owner.semanticsOwner

File: packages/flutter/test/rendering/rendering_tester.dart [hunk 6]
```diff
@@ -33,24 +33,22 @@ class TestRenderingFlutterBinding extends BindingBase with SchedulerBinding, Ser
...
       return;
-    if (SemanticsNode.hasListeners) {
-      pipelineOwner.flushSemantics();
-      if (phase == EnginePhase.flushSemantics)
-        return;
-      SemanticsNode.sendSemanticsTree();
-    }
+    pipelineOwner.flushSemantics();
+    if (phase == EnginePhase.flushSemantics)
+      return;
```
Comment (Hixie) [inline packages/flutter/test/rendering/rendering_tester.dart]: this changed the semantics. Now you're always sending the semantics tree. That's probably fine, but may affect some of the tests.

 (Also, there's no point having that last if test, better make it an assert.)

File: packages/flutter/lib/src/rendering/object.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: We should rename the `_SemanticsFragment.owner` field to `renderObjectOwner` and rename the RenderObject.owner field to `pipelineOwner`. Though I guess the latter will affect AbstractNode? hm.

Anyway that can all wait for some other patch. It's just weird seeing owner.owner.semanticsOwner.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/rendering/object.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/rendering/object.dart]: Done.

**Pull request merged**